examples/grpc-observability/frontend/src/main/java/net/devh/boot/grpc/examples/observability/frontend/FrontendApplication.java

```
@@ -0,0 +1,148 @@
+import net.devh.boot.grpc.examples.observability.proto.UnaryRequest;
+import net.devh.boot.grpc.examples.observability.proto.UnaryResponse;
+
+@SpringBootApplication
+public class FrontendApplication implements CommandLineRunner {
+
+    private static final Logger LOGGER = Logger.getLogger(FrontendApplication.class.getName());
+
+    public static void main(String[] args) {
+        SpringApplication.run(FrontendApplication.class, args);
+    }
+
+    @GrpcClient("backend")
+    private ExampleServiceStub stub;
+
+    private void CallUnaryRpc() {
+        byte[] bytes = new byte[ThreadLocalRandom.current().nextInt(10240, 20480)];
+        ThreadLocalRandom.current().nextBytes(bytes);
+        UnaryRequest request = UnaryRequest.newBuilder().setMessage(new String(bytes)).build();
+        stub.unaryRpc(request, new StreamObserver<>() {
+            @Override
+            public void onNext(UnaryResponse value) {}
+
+            @Override
+            public void onError(Throwable t) {
+                LOGGER.severe(Status.fromThrowable(t).toString());
+                CallUnaryRpc();
+            }
+
+            @Override
+            public void onCompleted() {
+                CallUnaryRpc();
+            }
+        });
+    }
+
+    private void CallClientStreamingRpc() {
+        byte[] bytes = new byte[ThreadLocalRandom.current().nextInt(10240, 20480)];
+        ThreadLocalRandom.current().nextBytes(bytes);
+        ClientStreamingRequest request = ClientStreamingRequest.newBuilder()
+                .setMessage(new String(bytes)).build();
+        StreamObserver<ClientStreamingRequest> requestStreamObserver = stub.clientStreamingRpc(
+                new StreamObserver<>() {
+                    @Override
+                    public void onNext(ClientStreamingResponse value) {}
+
+                    @Override
+                    public void onError(Throwable t) {
+                        CallClientStreamingRpc();
```

Is there a reason why are we starting another rpc onError() instead of logging the error?

---

On the client side, it just ignore the error and start the next call for testing.

---

I see. How about logging the error before starting the next call?